compiler/rustc_data_structures/src/tagged_ptr/copy/tests.rs

```diff
@@ -0,0 +1,61 @@
+use std::ptr;
+
+use crate::tagged_ptr::{CopyTaggedPtr, Pointer, Tag};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+enum Tag2 {
+    B00 = 0b00,
+    B01 = 0b01,
+    B10 = 0b10,
+    B11 = 0b11,
+}
+
+unsafe impl Tag for Tag2 {
+    const BITS: usize = 2;
+
+    fn into_usize(self) -> usize {
+        self as _
+    }
+
+    unsafe fn from_usize(tag: usize) -> Self {
+        const B00: usize = Tag2::B00 as _;
+        const B01: usize = Tag2::B01 as _;
+        const B10: usize = Tag2::B10 as _;
+        const B11: usize = Tag2::B11 as _;
+        match tag {
+            B00 => Tag2::B00,
+            B01 => Tag2::B01,
+            B10 => Tag2::B10,
+            B11 => Tag2::B11,
```

Comment:
```suggestion
        match tag {
            0b00 => Tag2::B00,
            0b01 => Tag2::B01,
            0b10 => Tag2::B10,
            0b11 => Tag2::B11,
```
this makes the match simpler and easier to understand

Reply:
Ehhh, I guess...

I prefer the version that uses casts, because it is resilient to discriminant changes (ideally I'd use `const { Tag2::B00 as _ }` as patterns but the feature is apparently incomplete)